From 7798ffb12e135cbb8237592764ac0608b1a7cd5e Mon Sep 17 00:00:00 2001
From: Sammy <r62914@freescale.com>
Date: Mon, 3 Aug 2009 14:44:08 +0800
Subject: [PATCH 107/505] ENGR00114742 vpu: Change test app for firmware
 upgrade to v1.2.0 on MX51

Change test app for firmware upgrade to v1.2.0 released on Jul.31.

Signed-off-by: Sammy He <r62914@freescale.com>
---
 test/mxc_vpu_test/enc.c | 16 +++++++++++++---
 1 file changed, 13 insertions(+), 3 deletions(-)

diff --git a/test/mxc_vpu_test/enc.c b/test/mxc_vpu_test/enc.c
index 73689ad..56400b5 100644
--- a/test/mxc_vpu_test/enc.c
+++ b/test/mxc_vpu_test/enc.c
@@ -583,10 +583,15 @@ encoder_configure(struct encode *enc)
 		else
 			ram_size = ((enc->picwidth + 15) & ~15) * 36 + 2048;
 		IOGetIramBase(&iram);
-		if ((iram.end - iram.start) < ram_size)
+		if ((iram.end - iram.start) < ram_size) {
+			err_msg("vpu iram is less than needed.\n");
+			return -1;
+		} else {
+			/* Allocate max iram for vpu encoder search ram*/
 			ram_size = iram.end - iram.start;
-		search_pa.searchRamAddr = iram.start;
-		search_pa.SearchRamSize = ram_size;
+			search_pa.searchRamAddr = iram.start;
+			search_pa.SearchRamSize = ram_size;
+		}
 	}
 
 	ret = vpu_EncGiveCommand(handle, ENC_SET_SEARCHRAM_PARAM, &search_pa);
@@ -709,6 +714,11 @@ encoder_open(struct encode *enc)
 	encop.mbReport = 0;
 	encop.mbQpReport = 0;
 	encop.rcIntraQp = -1;
+	encop.userQpMax = 0;
+	encop.userGamma = (Uint32)(0.75*32768);         /*  (0*32768 <= gamma <= 1*32768) */
+	encop.RcIntervalMode= 1;        /* 0:normal, 1:frame_level, 2:slice_level, 3: user defined Mb_level */
+	encop.MbInterval = 0;
+
 	encop.ringBufferEnable = 0;
 	encop.dynamicAllocEnable = 0;
 	encop.chromaInterleave = enc->cmdl->chromaInterleave;
-- 
1.8.0

